core/node/commitment_generator/src/lib.rs

```diff
@@ -335,29 +381,72 @@ impl CommitmentGenerator {
         }
     }
 
+    async fn next_batch_range(&self) -> anyhow::Result<Option<ops::RangeInclusive<L1BatchNumber>>> {
+        let mut connection = self
+            .connection_pool
+            .connection_tagged("commitment_generator")
+            .await?;
+        let Some(next_batch_number) = connection
+            .blocks_dal()
+            .get_next_l1_batch_ready_for_commitment_generation()
+            .await?
+        else {
+            return Ok(None);
+        };
+
+        let Some(last_batch_number) = connection
+            .blocks_dal()
+            .get_last_l1_batch_ready_for_commitment_generation()
+            .await?
+        else {
+            return Ok(None);
+        };
+        anyhow::ensure!(
+            next_batch_number <= last_batch_number,
+            "Unexpected node state: next L1 batch ready for commitment generation (#{next_batch_number}) is greater than \
+             the last L1 batch ready for commitment generation (#{last_batch_number})"
+        );
+        let last_batch_number =
+            last_batch_number.min(next_batch_number + self.parallelism.get() - 1);
+        Ok(Some(next_batch_number..=last_batch_number))
+    }
+
+    /// Runs this commitment generator indefinitely. It will process L1 batches added to the database
+    /// processed by the Merkle tree (or a tree fetcher), with a previously configured max parallelism.
     pub async fn run(self, stop_receiver: watch::Receiver<bool>) -> anyhow::Result<()> {
+        tracing::info!(
+            "Starting commitment generator with mode {:?} and parallelism {}",
+            self.commitment_mode,
+            self.parallelism
+        );
+        if self.connection_pool.max_size() < self.parallelism.get() {
```

Comment:
Main node runs CommitmentGenerator using a singleton connection pool. While it shouldn't affect anything as under normal operation it will process batches one by one, the log will be emitted. Does it make sense to provide larger pool or set parallelism=1?

Reply:
IMO, providing a larger pool would be a preferred option. Is this permissible from the infra perspective (i.e., is there a risk that this makes the DB to run out of connections)?